Userland/Libraries/LibGUI/ZoomableView.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022, Mustafa Quraish <mustafa@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ZoomableView.h"
+
+namespace GUI {
+
+constexpr float wheel_zoom_factor = 8.0f;
+
+void ZoomableView::set_scale(float new_scale)
+{
+    if (m_original_rect.is_null())
+        return;
+
+    m_scale = AK::clamp(new_scale, m_min_scale, m_max_scale);
```

Comment:
Do you really need the `AK::` prefix here?

Reply:
I think I've had this before: it's so that we *definitely* don't get the LibC version. Though I'm not sure whether this is the same.

Reply:
Don't think it was needed in this case, I've removed it.